dvc/remote/gs.py

```diff
@@ -136,16 +136,42 @@ def remove(self, path_info):
 
         blob.delete()
 
-    def _list_paths(self, bucket, prefix):
-        for blob in self.gs.bucket(bucket).list_blobs(prefix=prefix):
+    def _list_paths(self, bucket, prefix, max_items=None):
+        for blob in self.gs.bucket(bucket).list_blobs(
+            prefix=prefix, max_results=max_items
+        ):
             yield blob.name
 
     def list_cache_paths(self):
-        return self._list_paths(self.path_info.bucket, self.path_info.path)
+        for cache in self.walk_files(self.path_info):
+            yield cache.path
+
+    def walk_files(self, path_info):
+        for fname in self._list_paths(path_info.bucket, path_info.path):
+            # skip nested empty directories
+            if fname.endswith("/"):
+                continue
+            yield path_info.replace(fname)
+
+    def makedirs(self, path_info):
+        self.gs.bucket(path_info.bucket).blob(
+            (path_info / "").path
+        ).upload_from_string("")
```

Comment:
I might be missing some bigger picture here, but this will create a directory if it doesnot exist when checking out.

Reply:
@skshetry Very good point! So we won't "create" empty dirs inside of the added directory, as there won't be particular files to checkout in there. And the top level empty dir will be created anyway. That sounds like a great way to handle this without additional fuss.  CC @mroutis

Reply:
@efiop, I'm not sure if we should do that :thinking: 

I'd need to take a look, but so far, it looks like such compromise is not that bad, so yeah, maybe adding a `makedirs` should work.

Reply:
https://github.com/iterative/dvc/pull/2683

@skshetry, indeed, adding `makedirs` fixed the problem right away, wonderful catch!